cpp/src/prims/per_v_random_select_transform_outgoing_e.cuh

```diff
@@ -784,10 +802,7 @@ per_v_random_select_transform_outgoing_e(raft::handle_t const& handle,
                                          VertexFrontierBucketType const& frontier,
                                          EdgeSrcValueInputWrapper edge_src_value_input,
                                          EdgeDstValueInputWrapper edge_dst_value_input,
-#if 0  // FIXME: This will be necessary to include edge IDs in the output.
-       // Primitives API should be updated to support this in a consistent way.
-                               EdgeValueInputWrapper egde_value_input,
-#endif
+                                         EdgeValueInputWrapper egde_value_input,
```

Comment:
@ChuckHastings Heads-up on the required uniform neighbor sampling updates.

So, once this PR is merged, graph_t will no longer carry edge weights. `create_graph_from_edgelist` will return graph_t & renumber_map and optional `edge_property_t` carrying edge weights and additional `edge_property_t` carrying edge ID & type pairs.

If you want to include edge weight, ID, and types in the sampled output, you need to create a concatenated view from `edge_property_t` objects holding edge weights and ID & types (like you created a concatenated view in Louvain, but this time, for edge_property_t instead of edge_src|dst_property_t).

Then, in the `e_op` functor, you need to read the input edge property and return this.

Reply:
https://github.com/rapidsai/cugraph/pull/2843/files#diff-5df15be3808e4ae0b0243de76a597a37945dbb1e4bfe7014068eba8673aef4ffR273

So, now every `e_op` takes 5 input arguments (src, dst, src_property, dst_property, edge_property). This edge_property can be weight for traditional graph algorithms working on weighted graphs and for sampling, it can be a thrust::tuple of edge weight, ID, and type.

Reply:
So is the intent that the uniform neighbor sampling when this PR is merged will not support weights and we'll update it in a separate PR, or will those changes get added to this PR?

Reply:
This supports weights as before. But weights are passed as an edge property (which is separate from a graph) instead of being part of a graph.
See https://github.com/rapidsai/cugraph/pull/2843/files#diff-658e5c3027b9f03d9e11f87dd80b05d7e8da08cd17eb8be76360d0c75798461dR196

Later, we need to update an edge property to include not just weight, but edge ID & type as well.

Reply:
```
  if (edge_weight_view) {<= edge_weight_view here should be replaced with edge_(weight, ID, type)_view in the future.
    auto [sample_offsets, sample_e_op_results] = cugraph::per_v_random_select_transform_outgoing_e(
      handle,
      graph_view,
      vertex_frontier.bucket(0),
      edge_src_dummy_property_t{}.view(),
      edge_dst_dummy_property_t{}.view(),
      *edge_weight_view,
      sample_edges_op_t<vertex_t, weight_t>{},
      rng_state,
      fanout,
      with_replacement,
      std::make_optional<thrust::tuple<vertex_t, vertex_t, weight_t>>(
        cugraph::invalid_vertex_id<vertex_t>::value,
        cugraph::invalid_vertex_id<vertex_t>::value,
        weight_t{}),
      true);
    majors  = std::move(std::get<0>(sample_e_op_results));
    minors  = std::move(std::get<1>(sample_e_op_results));
    weights = std::move(std::get<2>(sample_e_op_results));
  } else {
    auto [sample_offsets, sample_e_op_results] = cugraph::per_v_random_select_transform_outgoing_e(
      handle,
      graph_view,
      vertex_frontier.bucket(0),
      edge_src_dummy_property_t{}.view(),
      edge_dst_dummy_property_t{}.view(),
      edge_dummy_property_t{}.view(),
      sample_edges_op_t<vertex_t, void>{},
      rng_state,
      fanout,
      with_replacement,
      std::make_optional<thrust::tuple<vertex_t, vertex_t>>(
        cugraph::invalid_vertex_id<vertex_t>::value, cugraph::invalid_vertex_id<vertex_t>::value),
      true);
    majors = std::move(std::get<0>(sample_e_op_results));
    minors = std::move(std::get<1>(sample_e_op_results));
  }
  ```

Reply:
>graph_t will no longer carry edge weights.
Maybe this confused you. graph_t will no longer carry edge weights but edge_property_t will carry edge weights, so we're still dealing with edge weights. It's just stored in a separate object.

Reply:
I'm sorry, I misunderstood your comment.  I understood the concept of the edge weights moving to `edge_property_t`, but I misinterpreted what you were saying in the comment.

Yes, we will now be able to update the sampling code to use a real edge id (instead of the hack of loading the edge id in the weight), and that will be done in a separate PR.  It will also require changes to PLC and python bindings to take advantage of the new feature.